reproman/resource/base.py

```diff
@@ -29,6 +29,63 @@
 lgr = logging.getLogger('reproman.resource.base')
 
 
+def discover_types():
+    """Discover resource types by inspecting the resource directory files.
+
+    Returns
+    -------
+    string list
+        List of resource identifiers extracted from file names.
+    """
+    l = []
+    for f in glob(op.join(op.dirname(__file__), '*.py')):
+        f_ = op.basename(f)
+        if f_ in ('base.py',) or f_.startswith('_'):
+            continue
+        l.append(f_[:-3].replace('_', '-'))
+    return sorted(l)
+
+
+def get_resource_class(name):
```

Comment:
There's a good amount of pure code movement here, so something like `git {log,show} --color-moved=dimmed_zebra --color-moved-ws=allow-indentation-change ...` might be helpful for reviewing.